From decc07b36db98a15982a84b09ff6cb709a320312 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Sat, 19 Apr 2025 19:52:49 +0300
Subject: [PATCH] [FSW-14755] fs1: update exports for new spi dma

With this patch, updated dma api declarations
and dma status enum based on new spi dma update.

Change-Id: Ifae1da4014062e783855cbcdd43c5fc5afca512b
---

diff --git a/fs/fs1/boot_rom_export_dev.h b/fs/fs1/boot_rom_export_dev.h
index f405193..608f1a9 100644
--- a/fs/fs1/boot_rom_export_dev.h
+++ b/fs/fs1/boot_rom_export_dev.h
@@ -18,6 +18,7 @@
 	DMA_UNALIGNED_IMG_SIZE,
 	DMA_UNALIGNED_DEST_ADDR,
 	DMA_INVALID_IMG_SIZE,
+	DMA_INVALID_FLASH_OFFSET, /* When flash offset + image size is beyond flash limit */
 	DMA_ERROR_MAX
 };
 
@@ -36,14 +37,10 @@
  * @param image_size size of the image to be dma from spi
  * @param flash_offset start offset of image in flash
  * @param dest_addr fw mem or hbw address where image should be copied
- * @param skip_crc_check api will skip crc check when true
- * @param manual_crc api will perform manual crc check at the end of dma when true
- * @param expected_crc 32-bit crc value to compare with at the end of dma. Valid
- * only if manual_crc is true.
+ * @param wait flasg to choose to wait or return after dma transfer start
  */
 enum dma_error boot_rom_spi_dma(uint32_t image_size, uint32_t flash_offset,
-				uint64_t dest_addr, bool skip_crc_check,
-				bool manual_crc, uint32_t expected_crc)__attribute__ ((long_call));
+	uint64_t dest_addr, bool wait);
 
 int secure_boot_verify_image_hash384(uint8_t *exp_hash, unsigned char *img,
 				uint32_t img_len) __attribute__ ((long_call));
